File: src/lib/api/reports.ts
import { invoke } from "@tauri-apps/api/core"

export type ReportRangeInput = {
  from: string | null
  to: string | null
  shift_id: number | null
}

export type SalesSummary = {
  revenue: number
  order_count: number
  avg_order_value: number
  discount_total: number
}

export type PaymentMixRow = {
  method: string
  total: number
  count: number
}

export type TopProductRow = {
  product_name: string
  quantity: number
  revenue: number
}

export const reportSalesSummary = (payload: ReportRangeInput) =>
  invoke<SalesSummary>("report_sales_summary", { payload })

export const reportPaymentMix = (payload: ReportRangeInput) =>
  invoke<PaymentMixRow[]>("report_payment_mix", { payload })

export const reportTopProducts = (payload: ReportRangeInput, limit = 10) =>
  invoke<TopProductRow[]>("report_top_products", { payload, limit })

export const reportDiscountTotal = (payload: ReportRangeInput) =>
  invoke<number>("report_discount_total", { payload })

export const reportShiftSummary = (shiftId: number) =>
  invoke<SalesSummary>("report_shift_summary", { shiftId })

File: src-tauri/src/domain/reports.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Deserialize)]
pub struct ReportRangeInput {
    pub from: Option<String>,
    pub to: Option<String>,
    pub shift_id: Option<i64>,
}

#[derive(Debug, Clone, Serialize)]
pub struct SalesSummary {
    pub revenue: i64,
    pub order_count: i64,
    pub avg_order_value: i64,
    pub discount_total: i64,
}

#[derive(Debug, Clone, Serialize)]
pub struct PaymentMixRow {
    pub method: String,
    pub total: i64,
    pub count: i64,
}

#[derive(Debug, Clone, Serialize)]
pub struct TopProductRow {
    pub product_name: String,
    pub quantity: i64,
    pub revenue: i64,
}
